add windows build to GitHub Actions

### Problem with SQLite 

Not possible to link to SQLite

- Needs to set: SQLITE3_LIB_DIR

Inststalled software: [Windows server 2019](https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md)

```bash
"msvcrt.lib" "/DLL" "/IMPLIB:D:\\a\\nafta\\nafta\\target\\debug\\deps\\migrations_macros-9cc352b842dbe7f2.dll.lib"
  = note: LINK : fatal error LNK1181: cannot open input file 'sqlite3.lib'
```

- [x] vcpkg install using [SQLite vcpkg]  (https://vcpkg.readthedocs.io/en/latest/examples/installing-and-using-packages/)

- [ ] check [diesel azure pipelines](https://github.com/diesel-rs/diesel/blob/1fe9487c92a73c26f26d3ce7e6924d2659101a34/azure-pipelines.yml#L110)

- [ ] try to use chocolatey and this https://github.com/diesel-rs/diesel/issues/487#issuecomment-349179729

- [x] try this approach https://github.com/diesel-rs/diesel/issues/487#issuecomment-444874556

- [ ] https://github.com/ruby/ruby/blob/master/.github/workflows/windows.yml#L53

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -28,14 +28,16 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
+    - name: Check vcvar64.bat ( remove when working )
+      run: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
     - name: Install SQLite using Chocolatey
       run: |
         choco install sqlite
-    - name: Create sqlite3.lib
-      run: |
-        cd C:\ProgramData\chocolatey\lib\SQLite\tools
-        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
-        lib /def:sqlite3.def /out:sqlite3.lib
+    - name: Run vcvar64.bat
+      run: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+    - name: Create sqlite3
+      run: "lib /def:sqlite3.def /out:sqlite3.lib"
+      working-directory: "C:\ProgramData\chocolatey\lib\SQLite\tools"
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -3,6 +3,7 @@ name: build
 on: [push]
 
 jobs:
+  # TODO: enable whent windows job is working
   # build-ubuntu:
   #   runs-on: ubuntu-latest
 
@@ -28,6 +29,8 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
+    env: 
+      SQLITE3_LIB_DIR: "C:\sqlite"
     - name: "Install SQLite"
       shell: cmd
       run: |
@@ -39,25 +42,9 @@ jobs:
           lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
           set PATH=%PATH%;C:\sqlite
           echo "##vso[task.setvariable variable=PATH;]%PATH%;C:\sqlite"
-
-    # - name: "Check vcvar64.bat ( remove when working )"
-    #   run: |
-    #     call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
-    #   shell: cmd
-
-    # - name: Install SQLite using Chocolatey
-    #   run: |
-    #     choco install sqlite
-
-    # - name: Run vcvar64.bat
-    #   run: |
-    #     call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
-    #     cd  "C:\ProgramData\chocolatey\lib\SQLite\tools"
-    #     lib /def:sqlite3.def /out:sqlite3.lib
-    #   shell: cmd
-
     - uses: actions/checkout@v1
-    - name: Build
-      run: cargo build 
+    # TODO: enable when test working
+    # - name: Build
+    #   run: cargo build 
     - name: Run tests
       run: cargo test

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -28,21 +28,33 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
-    - name: "Check vcvar64.bat ( remove when working )"
-      run: |
-        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+    - name: "Install SQLite"
       shell: cmd
-
-    - name: Install SQLite using Chocolatey
       run: |
-        choco install sqlite
+          mkdir C:\sqlite
+          CD /D C:\sqlite
+          curl -fsS --retry 3 --retry-connrefused -o sqlite3.zip https://sqlite.org/2017/sqlite-dll-win64-x64-3160200.zip
+          7z e sqlite3.zip -y
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          lib /machine:x64 /def:sqlite3.def /out:sqlite3.lib
+          set PATH=%PATH%;C:\sqlite
+          echo "##vso[task.setvariable variable=PATH;]%PATH%;C:\sqlite"
 
-    - name: Run vcvar64.bat
-      run: |
-        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
-        cd  "C:\ProgramData\chocolatey\lib\SQLite\tools"
-        lib /def:sqlite3.def /out:sqlite3.lib
-      shell: cmd
+    # - name: "Check vcvar64.bat ( remove when working )"
+    #   run: |
+    #     call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+    #   shell: cmd
+
+    # - name: Install SQLite using Chocolatey
+    #   run: |
+    #     choco install sqlite
+
+    # - name: Run vcvar64.bat
+    #   run: |
+    #     call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+    #     cd  "C:\ProgramData\chocolatey\lib\SQLite\tools"
+    #     lib /def:sqlite3.def /out:sqlite3.lib
+    #   shell: cmd
 
     - uses: actions/checkout@v1
     - name: Build

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -29,15 +29,21 @@ jobs:
     runs-on: windows-latest
     steps:
     - name: "Check vcvar64.bat ( remove when working )"
-      run: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+      shell: cmd
+
     - name: Install SQLite using Chocolatey
       run: |
         choco install sqlite
+
     - name: Run vcvar64.bat
-      run: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
-    - name: Create sqlite3
-      run: "lib /def:sqlite3.def /out:sqlite3.lib"
-      working-directory: "C:/ProgramData/chocolatey/lib/SQLite/tools"
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+        cd  "C:\ProgramData\chocolatey\lib\SQLite\tools"
+        lib /def:sqlite3.def /out:sqlite3.lib
+      shell: cmd
+
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -29,7 +29,10 @@ jobs:
     runs-on: windows-latest
     steps:
     - name: Install SQLite using Chocolatey
-      run: choco install sqlite
+      run: |
+        choco install sqlite
+        cd C:\ProgramData\chocolatey\lib\SQLite\tools
+        lib /def:sqlite3.def /out:sqlite3.lib
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -27,11 +27,9 @@ jobs:
 
   build-windows:
     runs-on: windows-latest
-    env:
-      VCPKGRS_DYNAMIC: 1
     steps:
-    - name: Install sqlite
-      run: vcpkg install sqlite3:x64-windows
+    - name: Install SQLite using Chocolatey
+      run: choco install sqlite
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -28,7 +28,7 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
-    - name: Check vcvar64.bat ( remove when working )
+    - name: "Check vcvar64.bat ( remove when working )"
       run: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
     - name: Install SQLite using Chocolatey
       run: |

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -3,7 +3,7 @@ name: build
 on: [push]
 
 jobs:
-  build:
+  build-ubuntu:
     runs-on: ubuntu-latest
 
     steps:
@@ -27,6 +27,8 @@ jobs:
 
   build-windows:
     runs-on: windows-latest
+    env:
+      VCPKGRS_DYNAMIC: 1
     steps:
     - name: Install sqlite
       run: vcpkg install sqlite3:x64-windows

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -3,27 +3,27 @@ name: build
 on: [push]
 
 jobs:
-  build-ubuntu:
-    runs-on: ubuntu-latest
+  # build-ubuntu:
+  #   runs-on: ubuntu-latest
 
-    steps:
-    # Needed because of: migrations_macros v1.4.2
-    # note: /usr/bin/ld: cannot find -lsqlite3
-    - name: Install sqlite
-      run: sudo apt-get install libsqlite3-dev
-    - uses: actions/checkout@v2
+  #   steps:
+  #   # Needed because of: migrations_macros v1.4.2
+  #   # note: /usr/bin/ld: cannot find -lsqlite3
+  #   - name: Install sqlite
+  #     run: sudo apt-get install libsqlite3-dev
+  #   - uses: actions/checkout@v2
 
-    - run: cargo build
-    - run: cargo test
+  #   - run: cargo build
+  #   - run: cargo test
 
-  build-macos:
-    runs-on: macos-latest
-    steps:
-    - uses: actions/checkout@v1
-    - name: Build
-      run: cargo build 
-    - name: Run tests
-      run: cargo test
+  # build-macos:
+  #   runs-on: macos-latest
+  #   steps:
+  #   - uses: actions/checkout@v1
+  #   - name: Build
+  #     run: cargo build 
+  #   - name: Run tests
+  #     run: cargo test
 
   build-windows:
     runs-on: windows-latest
@@ -31,6 +31,8 @@ jobs:
     - name: Install SQLite using Chocolatey
       run: |
         choco install sqlite
+    - name: Create sqlite3.lib
+      run: |
         cd C:\ProgramData\chocolatey\lib\SQLite\tools
         lib /def:sqlite3.def /out:sqlite3.lib
     - uses: actions/checkout@v1

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -29,7 +29,7 @@ jobs:
   build-windows:
     runs-on: windows-latest
     env: 
-      SQLITE3_LIB_DIR: "C:\sqlite"
+      SQLITE3_LIB_DIR: "C:\\sqlite"
     steps:
     - name: "Install SQLite"
       shell: cmd

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -34,6 +34,7 @@ jobs:
     - name: Create sqlite3.lib
       run: |
         cd C:\ProgramData\chocolatey\lib\SQLite\tools
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
         lib /def:sqlite3.def /out:sqlite3.lib
     - uses: actions/checkout@v1
     - name: Build

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -24,3 +24,12 @@ jobs:
       run: cargo build 
     - name: Run tests
       run: cargo test
+
+  build-windows:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build
+      run: cargo build 
+    - name: Run tests
+      run: cargo test

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -28,6 +28,8 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
+    - name: Install sqlite
+      run: vcpkg install sqlite3:x64-windows
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -3,28 +3,27 @@ name: build
 on: [push]
 
 jobs:
-  # TODO: enable whent windows job is working
-  # build-ubuntu:
-  #   runs-on: ubuntu-latest
+  build-ubuntu:
+    runs-on: ubuntu-latest
 
-  #   steps:
-  #   # Needed because of: migrations_macros v1.4.2
-  #   # note: /usr/bin/ld: cannot find -lsqlite3
-  #   - name: Install sqlite
-  #     run: sudo apt-get install libsqlite3-dev
-  #   - uses: actions/checkout@v2
+    steps:
+    # Needed because of: migrations_macros v1.4.2
+    # note: /usr/bin/ld: cannot find -lsqlite3
+    - name: Install sqlite
+      run: sudo apt-get install libsqlite3-dev
+    - uses: actions/checkout@v2
 
-  #   - run: cargo build
-  #   - run: cargo test
+    - run: cargo build
+    - run: cargo test
 
-  # build-macos:
-  #   runs-on: macos-latest
-  #   steps:
-  #   - uses: actions/checkout@v1
-  #   - name: Build
-  #     run: cargo build 
-  #   - name: Run tests
-  #     run: cargo test
+  build-macos:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build
+      run: cargo build 
+    - name: Run tests
+      run: cargo test
 
   build-windows:
     runs-on: windows-latest
@@ -43,8 +42,9 @@ jobs:
           set PATH=%PATH%;C:\sqlite
           echo "##vso[task.setvariable variable=PATH;]%PATH%;C:\sqlite"
     - uses: actions/checkout@v1
-    # TODO: enable when test working
-    # - name: Build
-    #   run: cargo build 
+    - name: Build
+      run: cargo build 
     - name: Run tests
+      env: 
+        SQLITE3_LIB_DIR: "C:\\sqlite"
       run: cargo test

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -28,9 +28,9 @@ jobs:
 
   build-windows:
     runs-on: windows-latest
-    steps:
     env: 
       SQLITE3_LIB_DIR: "C:\sqlite"
+    steps:
     - name: "Install SQLite"
       shell: cmd
       run: |

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -29,15 +29,15 @@ jobs:
     runs-on: windows-latest
     steps:
     - name: "Check vcvar64.bat ( remove when working )"
-      run: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+      run: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
     - name: Install SQLite using Chocolatey
       run: |
         choco install sqlite
     - name: Run vcvar64.bat
-      run: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+      run: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat"
     - name: Create sqlite3
       run: "lib /def:sqlite3.def /out:sqlite3.lib"
-      working-directory: "C:\ProgramData\chocolatey\lib\SQLite\tools"
+      working-directory: "C:/ProgramData/chocolatey/lib/SQLite/tools"
     - uses: actions/checkout@v1
     - name: Build
       run: cargo build 

